client/verta/verta/deployment/_deployedmodel.py

```diff
@@ -183,43 +169,63 @@ def get_curl(self):
         """
         headers = self.headers()
         headers.update({"Content-Type": "application/json"})
-        curl = "curl -X POST {} -d '' -H \"Content-Type: application/json\"".format(
-            self.prediction_url
-        )
+        curl = f"curl -X POST {self.prediction_url} -d '' -H \"Content-Type: application/json\""
         for header, value in headers.items():
-            curl += ' -H "{}: {}" '.format(header, value)
+            curl += f" -H \"{header}: {value}\""
         return curl
 
+    # TODO: Removed deprecated `always_retry_404` and `always_retry_429` params
     def predict(
         self,
-        x,
+        x: List[Any],
         compress=False,
-        max_retries=5,
-        always_retry_404=True,
-        always_retry_429=True,
-    ):
+        prediction_id: Optional[str] = None,
```

Comment:
For API compatibility, new parameters should be added after all the existing ones. Otherwise, this breaks user code that calls

```python
deployed_model.predict(X, False, 20)
```

I'm not wholly opposed to shipping API breakages, but I think we'd like to avoid it here for at least a release or two.

Admittedly, this is frustrating because it becomes difficult to group related parameters together.

Reply:
For extra fun, might I float the idea of implementing the new parameters as [keyword-only arguments](https://peps.python.org/pep-3102/), which frees them in the future from this positional compatibility concern and I think is how they will generally be passed anyway (e.g. in the tests you've written).

It's something I've always wanted to use after we dropped Python 2, but I'm not sure how common they are in the Python ecosystem